parsers/NTESMO.py

```diff
@@ -135,6 +146,11 @@ def parse_consumption(
     price: bool = False,
 ) -> List[dict]:
     data_points = []
+
```

Comment:
Since in this function the `target_datetime` cannot be None according to the type hints I don't think the datetime should be set here. It should rather be set at the fetch_production/consumption/price level.